core/example/benchmark/Cabana_CommPerformance.cpp

```diff
@@ -0,0 +1,604 @@
+ * Copyright (c) 2018-2019 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cabana_Core.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+#include <chrono>
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <numeric>
+#include <fstream>
+
+#include <mpi.h>
+
+//---------------------------------------------------------------------------//
+// Parallel timer. Carries multiple data points for each timer to allow for
+// parametric sweeps. Each timer can do multiple runs over each data point in
+// the parameter sweep. The name of the data point and its values can then be
+// injected into the output table.
+class ParallelTimer
+{
+  public:
+
+    // Create the timer.
+    ParallelTimer( MPI_Comm comm,
+                   const std::string& name,
+                   const int num_data )
+        : _comm( comm )
+        , _name( name )
+        , _starts( num_data )
+        , _data( num_data )
+        , _is_stopped( num_data, true )
+    {}
+
+    // Start the timer for the given data point. It is assumed that this
+    // function is called collectively across all ranks in the communicator.
+    void start( const int data_point )
+    {
+        if ( !_is_stopped[data_point] )
+            throw std::logic_error( "attempted to start a running timer" );
+        auto now = std::chrono::high_resolution_clock::now();
+        _starts[data_point].push_back( now );
+        _is_stopped[data_point] = false;
+    }
+
+    // Stop the timer at the given data point. It is assumed that this
+    // function is called collectively across all ranks in the communicator.
+    void stop( const int data_point )
+    {
+        if ( _is_stopped[data_point] )
+            throw std::logic_error( "attempted to stop a stopped timer" );
+        auto now = std::chrono::high_resolution_clock::now();
+        _data[data_point].push_back( std::chrono::duration_cast<std::chrono::microseconds>(
+                                         now - _starts[data_point].back() ) );
+        _is_stopped[data_point] = true;
+    }
+
+    // Get the number of timer data points.
+    int numDataPoint() const
+    {
+        return _data.size();
+    }
+
+    // Get the number of timer runs for a given data point.
+    int numRun( const int data_point ) const
+    {
+        return _data[data_point].size();
+    }
```

Comment:
I would have dropped it since it is not call anywhere.  Anyway no big deal

Reply:
removed